mcs/class/System/Test/System.Diagnostics/ProcessTest.cs

```diff
@@ -1085,5 +1085,31 @@ public void TestDoubleBeginErrorReadLine ()
 			}
 		}
 #endif // MONO_FEATURE_PROCESS_START
+
+#if MONO_FEATURE_PROCESS_START
+		[Test]
+		[NUnit.Framework.Category ("MobileNotWorking")]
+		public void TestExitedRaisedTooSoon ()
+		{
+			switch (Environment.OSVersion.Platform) {
+			case PlatformID.Unix:
+			case PlatformID.MacOSX:
+				break;
+			default:
+				Assert.Ignore ("no sleep command on windows");
+				return;
+			}
+
+			var sleeptime = 5;
+			var mre = new ManualResetEvent (false);
+			var process = Process.Start("sleep", sleeptime.ToString ());
```

Comment:
I think we can generic test utility to launch any available external process somewhere